### ISSUE TITLE
backports.ssl_match_hostname missing

### ISSUE THREAD

**Issue by mscook** [rethinkdb/rethinkdb#4683]
Consider adding backports.ssl_match_hostname missing to requirements.txt:

Current situation after pip install rethinkdb:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mscook/.virtualenvs/RethinkDB_2.1.0/local/lib/python2.7/site-packages/rethinkdb/__init__.py", line 3, in <module>
    from .net import *
  File "/home/mscook/.virtualenvs/RethinkDB_2.1.0/local/lib/python2.7/site-packages/rethinkdb/net.py", line 33, in <module>
    from backports.ssl_match_hostname import match_hostname, CertificateError
ImportError: No module named backports.ssl_match_hostname
```

Install it:

```
pip install backports.ssl_match_hostname
Collecting backports.ssl-match-hostname
/home/mscook/.venvburrito/lib/python2.7/site-packages/pip-7.1.0-py2.7.egg/pip/_vendor/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading backports.ssl_match_hostname-3.4.0.2.tar.gz
Building wheels for collected packages: backports.ssl-match-hostname
  Running setup.py bdist_wheel for backports.ssl-match-hostname
  Stored in directory: /home/mscook/.cache/pip/wheels/81/3d/36/f3c726a42caed585046d710e0e450bda1075c61c1c43443831
Successfully built backports.ssl-match-hostname
Installing collected packages: backports.ssl-match-hostname
Successfully installed backports.ssl-match-hostname-3.4.0.2
```

Works:

```
Python 2.7.3 (default, Mar 18 2014, 05:13:23)
[GCC 4.6.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import rethinkdb a r
  File "<stdin>", line 1
    import rethinkdb a r
                     ^
SyntaxError: invalid syntax
>>> import rethinkdb as r
>>>
```

**Comment by danielmewes** [comment]
Thanks for opening an issue about this @mscook .

@AtnNn or @larkost could one of you pick this please?

**Comment by dpetrovics** [comment]
+1

**Comment by AtnNn** [comment]
`ssl_match_hostname` was added by @larkost:

https://github.com/rethinkdb/rethinkdb/tree/next/drivers/python/rethinkdb/backports/ssl_match_hostname

I'm not sure what is going on here, Maybe there is a name conflict. Maybe the files aren't being packaged correctly.

**Comment by sebastianneubauer** [comment]
+1 same problem for me with newest version 2.1.0. Worked with 2.0.0-2. BTW, where can one lookup the git revision corresponding to the package version on pypi? In git head, the version still claims it is 1.16.0-2...

**Comment by danielmewes** [comment]
@sebastianneubauer you should check the branch `v2.1.x`. `next` has the same driver, but the version isn't updated in there.
Can you try installing installing the module manually? `pip install backports.ssl_match_hostname`

**Comment by larkost** [comment]
It looks like the `backports` folder is not being included when we wrap things up for `pip` deploy. I will see what I can do to fix this.

**Comment by larkost** [comment]
I have what looks like a fix (I don't know how to completely simulate `pip` locally) in branch `larkost/4683-backports.ssl_match_hostname` (CR 3164). It turns out that binary distributions (bdist and bdist_egg) do not include any sub-folders unless they are in the `package_data` and `package_dir` entries in `setup.py`. Whereas sdist does not require this for inclusion.

**Comment by larkost** [comment]
This has been merged into `next` with e2a32acc and `v2.1.x` with 50092203.

**Comment by mgingras** [comment]
@larkost Just hit this using `pip install rethinkdb`

**Comment by larkost** [comment]
@mgingras: @danielmewes's response above about using `pip install backports.ssl_match_hostname` should be a good work-around until we get the new version out with this fix (and a couple of others). I have just been told that pre-build testing will begin in about a half hour, so that makes a release likely tomorrow or on Monday.

**Comment by mgingras** [comment]
@larkost Thanks!

**Comment by danielmewes** [comment]
The fix for this shipped in Python driver version `2.1.0.post1`.
